timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionDetectionHelper.java

```diff
@@ -0,0 +1,120 @@
+import org.junit.runners.model.Statement;
+
+public final class TimeLockCorruptionDetectionHelper implements TestRule {
+    private TimeLockCorruptionTestSetup timeLockCorruptionTestSetup = new TimeLockCorruptionTestSetup();
+
+    Set<PaxosValue> writeLogsOnDefaultLocalServer(int start, int end) {
+        return writeLogsOnServer(timeLockCorruptionTestSetup.getDefaultLocalServer(), start, end);
+    }
+
+    void writeLogsOnDefaultLocalAndRemote(int startingLogSeq, int latestLogSequence) {
+        writeLogsOnLocalAndRemote(
+                timeLockCorruptionTestSetup.getDefaultServerList(), startingLogSeq, latestLogSequence);
+    }
+
+    Set<PaxosValue> writeLogsOnServer(StateLogComponents server, int start, int end) {
+        return PaxosSerializationTestUtils.writeToLogs(server.acceptorLog(), server.learnerLog(), start, end);
+    }
+
+    void writeLogsOnLocalAndRemote(List<StateLogComponents> servers, int startingLogSeq, int latestLogSequence) {
+        servers.stream().forEach(server -> writeLogsOnServer(server, startingLogSeq, latestLogSequence));
+    }
+
+    StateLogComponents getDefaultLocalServer() {
+        return timeLockCorruptionTestSetup.getDefaultLocalServer();
+    }
+
+    List<StateLogComponents> getDefaultRemoteServerList() {
+        return timeLockCorruptionTestSetup.getDefaultRemoteServerList();
+    }
+
+    void induceGreaterAcceptedValueCorruptionOnDefaultLocalServer(int corruptSeq) {
+        induceGreaterAcceptedValueCorruption(timeLockCorruptionTestSetup.getDefaultLocalServer(), corruptSeq);
+    }
+
+    void induceGreaterAcceptedValueCorruption(StateLogComponents server, int corruptSeq) {
+        PaxosSerializationTestUtils.writeAcceptorStateForLogAndRound(
+                server.acceptorLog(),
+                corruptSeq,
+                Optional.of(PaxosSerializationTestUtils.createPaxosValueForRoundAndData(corruptSeq, corruptSeq + 1)));
+    }
+
+    List<CompletePaxosHistoryForNamespaceAndUseCase> getHistory() {
+        return timeLockCorruptionTestSetup.getPaxosLogHistoryProvider().getHistory();
+    }
+
+    void assertNoCorruptionViolations() {
+        assertViolationsDetectedForNamespaceAndUseCases(ImmutableSet.of(), ImmutableSet.of());
+    }
+
+    void assertAcceptedValueGreaterThanLearnedValue() {
+        assertViolationsDetectedForNamespaceAndUseCases(
+                ImmutableSet.of(CorruptionCheckViolation.ACCEPTED_VALUE_GREATER_THAN_LEARNED),
+                ImmutableSet.of(Constants.DEFAULT_NAMESPACE_AND_USE_CASE));
+    }
+
+    void assertViolationsDetected(Set<CorruptionCheckViolation> detectedViolations) {
+        assertViolationsDetectedForNamespaceAndUseCases(
+                detectedViolations, ImmutableSet.of(Constants.DEFAULT_NAMESPACE_AND_USE_CASE));
+    }
+
+    void assertViolationsDetectedForNamespaceAndUseCases(
+            Set<CorruptionCheckViolation> detectedViolations,
+            Set<NamespaceAndUseCase> namespaceAndUseCasesWithViolation) {
+
```

Comment:
nit: remove empty line
```suggestion
```

Reply:
With this method, either 1 or 0 detected violations are passed, which I'm not a fan of at all. I'd prefer if you separate out to two methods: one that takes a single violation and confirms it is there, and a second where the logic is essentially:
```java
assertThat(getViolationsToNamespaceToUseCaseMultimap()).isEmpty();
```